src/popup/router/index.ts

```diff
@@ -61,14 +64,20 @@ RouteQueryActionsController.init(router);
 
 const unbind = router.beforeEach(async (to, from, next) => {
   await watchUntilTruthy(() => store.state.isRestored);
-  next(
-    (
-      !RUNNING_IN_POPUP
-      && to.name === ROUTE_INDEX
-      && ((await browser?.storage.local.get(lastRouteKey)) as any)[lastRouteKey]
-    )
-    || undefined,
-  );
+
+  // This check is run to avoid unnecessary fetching from WalletStorage
+  if (RUNNING_IN_POPUP || to.name !== ROUTE_INDEX) {
+    next(undefined);
```

Comment:
In this case it will still await the `lastRoute` afterwards.

Reply:
I reorganised code that @pstachel did by creating separate lib. Can you please take a look again?